src/game/Objects/Unit.cpp

```diff
@@ -379,24 +383,17 @@ bool Unit::UpdateMeleeAttackingState()
 
             if (IsAttackReady(BASE_ATTACK))
             {
-                // prevent base and off attack in same time, delay attack at 0.2 sec
-                if (HaveOffhandWeapon())
-                {
-                    if (GetAttackTimer(OFF_ATTACK) < ATTACK_DISPLAY_DELAY)
-                        SetAttackTimer(OFF_ATTACK, ATTACK_DISPLAY_DELAY);
-                }
                 AttackerStateUpdate(pVictim, BASE_ATTACK);
-                ResetAttackTimer(BASE_ATTACK);
+                ResetAttackTimer(BASE_ATTACK, !m_openerAttack);
+                if (m_openerAttack && HaveOffhandWeapon() && GetAttackTimer(OFF_ATTACK) < ATTACK_DISPLAY_DELAY)
+                    SetAttackTimer(OFF_ATTACK, ATTACK_DISPLAY_DELAY);
+                m_openerAttack = false;
             }
-            if (HaveOffhandWeapon() && IsAttackReady(OFF_ATTACK))
+            else if (HaveOffhandWeapon() && IsAttackReady(OFF_ATTACK))

```

Comment:
Wouldn't this change cause off hand attack to never go off when server has high update time and main hand attack speed is lower than it?

Reply:
Yes it would in the case where the server update time is longer than mainhand attack speed. It probably shouldn't happen, but I have no retail wow videos of a DW unit attacking when the server is that laggy.

Note though "DW never hits if the server update time is longer than MH attack speed" is already true for how vmangos is before this PR, because of this:
```
                if (HaveOffhandWeapon())
                {
                    if (GetAttackTimer(OFF_ATTACK) < ATTACK_DISPLAY_DELAY)
                        SetAttackTimer(OFF_ATTACK, ATTACK_DISPLAY_DELAY);
                }
```
Making every world update when its very high update time, the mainhand lands and delays the offhand by 200 ms so it's no longer ready within that same update, then the next update happens and, the MH is ready so it attacks with it and delays the offhand by 200ms again, and so on making the offhand never hit.

I'm not sure what the retail wow would do in this situation. I don't have enough data to conclude if an offhand can hit in the same world update as a mainhand. People in classic discords describe a situation where if offhand and mainhand hit very close then the warriors with the talent flurry only lose one charge of the buff. Is that a spellbatching thing or were melee attacks batched as well in classic era, I don't know.